### ISSUE TITLE
shortcut rebase fixup

### ISSUE THREAD

**Issue by haoweicrushu** [jesseduffield/lazygit#387]
**Describe the bug**
Can't use f keyword to fixup commit below

**To Reproduce**
```
git init

echo a > a.txt
git add a.txt
git commit -m "commit 1"

echo b > b.txt
git add b.txt
git commit -m "commit 2"

echo c > c.txt
git add c.txt
git commit -m "commit 3"
```
1. start lazygit
2. press f to fixedup commit 3 to commit 2

**Screenshots**
<img width="1012" alt="Screen Shot 2019-03-11 at 8 56 37 AM" src="https://user-images.githubusercontent.com/5827907/54094410-a401f000-43db-11e9-83dc-e35151760f7c.png">
After press f
<img width="1011" alt="Screen Shot 2019-03-11 at 8 56 51 AM" src="https://user-images.githubusercontent.com/5827907/54094413-a7957700-43db-11e9-8e9f-7135bb72e6a2.png">

I follow direction typing ```git rebase --edit-todo``` and save

<img width="1013" alt="Screen Shot 2019-03-11 at 9 01 22 AM" src="https://user-images.githubusercontent.com/5827907/54094513-710c2c00-43dc-11e9-85e1-4f3a9fd50157.png">

<img width="744" alt="Screen Shot 2019-03-11 at 9 01 38 AM" src="https://user-images.githubusercontent.com/5827907/54094517-79fcfd80-43dc-11e9-900a-7314b8da5d4c.png">

still not work

**Expected**
rebase commit 3 to commit 2

like below
<img width="1012" alt="Screen Shot 2019-03-11 at 8 58 05 AM" src="https://user-images.githubusercontent.com/5827907/54094421-ceec4400-43db-11e9-8685-871e6dc0fd6a.png">

It worked on version 0.6

**Desktop :**
 - OS: MacOS 10.14.3
 - Lazygit Version 0.72

**Comment by jesseduffield** [comment]
I'm unable to replicate this issue locally. If I'm following you correctly, you open lazygit, stage the unstaged file, commit it in a commit named 'commit 3', then select the commit and press 'f' to do a fixup?

I was able to do this without any errors. I also tried doing it via an interactive rebase by pressing 'e' on commit 2, then 'f' on commit 3, them 'm' and selecting 'continue'.

Have I followed your reproduction steps correctly?

**Comment by haoweicrushu** [comment]
> I'm unable to replicate this issue locally. If I'm following you correctly, you open lazygit, stage the unstaged file, commit it in a commit named 'commit 3', then select the commit and press 'f' to do a fixup?
> 
Yes sir. I wanna press f to do a fixup. 

> I was able to do this without any errors. I also tried doing it via an interactive rebase by pressing 'e' on commit 2, then 'f' on commit 3, them 'm' and selecting 'continue'.
> 
I test this step, it works.
> Have I followed your reproduction steps correctly?

Press f to do a fixup is still available on version 0.72 ? 

Your steps is correct ! 

I can do a rebase with them.

**Comment by jesseduffield** [comment]
the error message seems to suggest that you're pressing 'f' on commit 1 which does not have a previous commit. But you're pressing 'f' on commit 3, is that correct? Are you able to show what your `.git/rebase-merge/git-rebase-todo` contains when that error appears?

**Comment by haoweicrushu** [comment]
>Are you able to show what your .git/rebase-merge/git-rebase-todo contains when that error appears?
>
Yes sir, I record a video to replicate.
[video](https://youtu.be/j-dzbD5q4nQ)(The quality will be better later )

On another Mac
git version 2.21.0
MacOS 10.14.2 Mojave
go version 1.12
lazygit version 0.7.2

**Comment by haoweicrushu** [comment]
fix in v0.8

**Comment by jesseduffield** [comment]
good to hear!
